lib/scholar/cluster/gmm.ex

```diff
@@ -0,0 +1,407 @@
+defmodule Scholar.Cluster.GaussianMixture do
+  @moduledoc """
+  Gaussian Mixture Model.
+
+  Gaussian Mixture model is a probabilistic model that assumes every data point is generated
+  by randomly choosing one of the several fixed Gaussian distributions and then sampling from it.
+  Its parameters are estimated using the Expectation-Maximization (EM) algorithm, which is an
+  iterative algorithm alternating between the two steps: the E-step which computes the
+  expectation of the Gaussian assignment for each data point x and the M-step which updates the
+  parameters to maximize the expectations found in E-step. While every iteration of the algorithm
+  is guaranteed to improve the log-likelihood, the final result depends on the initial values of
+  the parameters and the entire procedure should be repeated several times.
+
+  References:
+
+  * [1] - Mixtures of Gaussians and the EM algorithm https://cs229.stanford.edu/notes2020spring/cs229-notes7b.pdf
+  """
+
+  import Nx.Defn
+  import Scholar.Shared
+  require Nx
+
+  @derive {Nx.Container, containers: [:weights, :means, :covariances, :precisions_cholesky]}
+  defstruct [:weights, :means, :covariances, :precisions_cholesky]
+
+  opts = [
+    num_gaussians: [
+      required: true,
+      type: :pos_integer,
+      doc: "The number of Gaussian components in the mixture."
+    ],
+    num_runs: [
+      type: :pos_integer,
+      default: 1,
+      doc: "The number of initializations to perform."
```

Comment:
Not sure if "initializations" is the best word because I assume we're talking about running the full procedure and not just initializing it, right?

Reply:
Yes, running the full procedure but with different initial parameters. Indeed, word initializations might be misleading here.

Reply:
I guess "The number of times to initialize parameters and run the entire EM algorithm." might be a better description here.

Reply:
Agreed. This description will be maximally informative.